File: app/src/main/java/com/example/octodroid/data/SessionManager.java
package com.example.octodroid.data;

import android.content.Context;

import com.example.octodroid.data.prefs.SessionPrefs;
import com.example.octodroid.data.prefs.SessionPrefsSchema;

public class SessionManager {
    public static boolean isLoggedIn(Context context) {
        SessionPrefs prefs = SessionPrefsSchema.get(context);
        return prefs.hasUsername() && prefs.hasPassword();
    }

    public static void login(Context context) {
        SessionPrefs prefs = SessionPrefsSchema.get(context);
        GitHub.client().authorization(prefs.getUsername(), prefs.getPassword());
    }

    public static void logout(Context context) {
        SessionPrefs.get(context).clear();
    }
}

File: app/src/main/kotlin/com/example/octodroid/data/SessionManager.kt
package com.example.octodroid.data

import android.content.Context
import com.example.octodroid.data.prefs.SessionPrefs
import com.example.octodroid.data.prefs.SessionPrefsSchema

object SessionManager {
    fun isLoggedIn(context: Context): Boolean {
        val prefs = SessionPrefs.get(context)
        return prefs.hasUsername() && prefs.hasPassword()
    }

    fun login(context: Context) {
        val prefs = SessionPrefs.get(context)
        GitHub.client().authorization(prefs.username, prefs.password)
    }

    fun logout(context: Context) {
        SessionPrefs.get(context).clear()
    }
}
